### PR TITLE
refactor: complete purifying Response module

### LINKED ISSUE #2974: Clean `Response` module off `Wai` leftovers
> I expected the `Wai` module usage to go away from Response after this change, but it's still in some places:

https://github.com/PostgREST/postgrest/blob/a6e3eda5b20817f134beb8d660029bd2875c280f/src/PostgREST/Response.hs#L284-L290

https://github.com/PostgREST/postgrest/blob/a6e3eda5b20817f134beb8d660029bd2875c280f/src/PostgREST/Response.hs#L310-L316

> I don't see an easy fix for those now so LGTM.

_Originally posted by @steve-chavez in https://github.com/PostgREST/postgrest/issues/2971#issuecomment-1737537127_

### PR DESCRIPTION
**Pull request by develop7**
fixes #2974

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
  - #2846, Fix error when requesting `Prefer: count=<type>` and doing null filtering on embedded resources - @laurenceisla
  - #2959, Fix setting `default_transaction_isolation` unnecessarily - @steve-chavez
  - #2929, Fix arrow filtering on RPC returning dynamic TABLE with composite type - @steve-chavez
+ - #2974, Clean `Response` module off `Wai` leftovers - @develop7
```
Comment (steve-chavez) [inline CHANGELOG.md:28]: Oh, we don't keep refactors on the CHANGELOG. Only feats/fixes/breaking changes.
```suggestion
```

**Comment by steve-chavez** [comment]
styler is failing, see https://github.com/PostgREST/postgrest/actions/runs/6353379225/job/17257990872?pr=2976

**Comment by develop7** [comment]
> styler is failing, see

whoops, sorry, done & done

File: CHANGELOG.md (hunk 1 above)
Comment (steve-chavez) [reply CHANGELOG.md:28]: @develop7 This one still remains

Comment (develop7) [reply CHANGELOG.md]: yup, not anymore

**Pull request merged**